### DISCUSSION TITLE
Discord RPC doesn't show any game data

### DISCUSSION THREAD

**Discussion by nzxl101** [ppy/osu#20821] [Q&A]
As the title says, the game shows up fine in Discord but doesn't contain any information about the current beatmap, ruleset, or player.
This happens in both lazer and stable.

<table>
    <thead>
        <th>✅ Expected behavior</th>
        <th>❌</th>
    </thead>
    <tbody>
        <tr>
            <td>
                <img src="https://user-images.githubusercontent.com/63413558/196583504-57e97cad-3532-4e0c-9238-27c952fe574f.png">
            </td>
            <td>
                <img src="https://user-images.githubusercontent.com/63413558/196583424-c0cd60b1-8cfa-42f9-a462-672438c47017.png">
            </td>
        </tr>
    </tbody>
</table>

Potential solutions I've tried:
- Restart PC ❌ - doesn't work
- Uninstall Discord and Reinstall ❌ - doesn't work
- Use Discord PTB ❌ - doesn't work
- Run Discord as Non-Admin and Admin ❌ - both don't work
- Run Lazer as Admin ✅ - about a 80% chance that it connects, but then Tablet drivers stop working
- Run Stable as Admin ❌

Reported data from Discord API:
```
osu! stable:
Activity {
    id: 'e5f01207cf373bb1',
    name: 'osu!',
    type: 'PLAYING',
    url: null,
    details: null,
    state: null,
    applicationId: '367827983903490050',
    timestamps: { start: 2022-10-19T02:23:43.089Z, end: null },
    syncId: null,
    platform: null,
    party: null,
    assets: null,
    flags: ActivityFlags { bitfield: 0 },
    emoji: null,
    sessionId: null,
    buttons: [],
    createdTimestamp: 1666146241419
}

osu! lazer:
Activity {
    id: 'e5f01207cf373bb1',
    name: 'osu!',
    type: 'PLAYING',
    url: null,
    details: null,
    state: null,
    applicationId: '367827983903490050',
    timestamps: { start: 2022-10-19T02:24:23.089Z, end: null },
    syncId: null,
    platform: null,
    party: null,
    assets: null,
    flags: ActivityFlags { bitfield: 0 },
    emoji: null,
    sessionId: null,
    buttons: [],
    createdTimestamp: 1666146281399
}
```

https://user-images.githubusercontent.com/63413558/196587956-980a1084-9052-4e47-8e25-c0ea57b07d33.mp4

[runtime.log](https://github.com/ppy/osu/files/9816393/runtime.log)

This only happens with osu! I don't have any RPC issues on other games.
It would be nice to hear from others who may also have the same problem.

**Comment by smoogipoo** [comment]
Don't run osu!lazer or osu!stable as admin. Otherwise, I can't repro this on Linux, so if anything it might be Windows-specific (potentially even W11-specific). Is anyone else able to repro?

**Reply by peppy** [reply]
I already tested on windows and both stable/lazer work fine here.

**Reply by nzxl101** [reply]
I've tested it on a completely different pc with a fresh install of Windows 10 + Newest Updates installed just to make sure it's not a specific user config issue in the OS and it also doesn't show the Rich Presence.
![image](https://user-images.githubusercontent.com/63413558/196599873-5a7636e5-a3c9-4a95-b6c2-907212dc4186.png)

**Reply by integerrr** [reply]
Just a thought, have you actually set the corresponding DRP settings to `Full` in game?
![image](https://user-images.githubusercontent.com/7279624/196722116-f2bf6861-96a2-4f8c-9a5f-f7a23acef6a0.png)

**Reply by agatemosu** [reply]
Happens to me only in osu!(stable), not in osu!.

**Reply by nzxl101** [reply]
> Just a thought, have you actually set the corresponding DRP settings to `Full` in game? ![image](https://user-images.githubusercontent.com/7279624/196722116-f2bf6861-96a2-4f8c-9a5f-f7a23acef6a0.png)

Yes, It's on full.

**Reply by Vortex100K** [reply]
> Don't run osu!lazer or osu!stable as admin. Otherwise, I can't repro this on Linux, so if anything it might be Windows-specific (potentially even W11-specific). Is anyone else able to repro?

Im on win11 ill try and repro
